lib/LMS.class.php

```diff
@@ -1582,75 +1582,146 @@ public function UpdateRegisterData($name, $url, $hidden)
-                $params['username'] = (!$user ? $smtp_username : $user);
-                $params['password'] = (!$pass ? ConfigHelper::getConfig('mail.smtp_password') : $pass);
-            } else
-                $params['auth'] = false;
-            $params['persist'] = $persist;
-
-            $error = $this->mail_object = & Mail::factory('smtp', $params);
-            //if (PEAR::isError($error))
-            if (is_a($error, 'PEAR_Error'))
-                return $error->getMessage();
-        }
-
-        $headers['X-Mailer'] = 'LMS-' . $this->_version;
-        if (!empty($_SERVER['REMOTE_ADDR']))
-            $headers['X-Remote-IP'] = $_SERVER['REMOTE_ADDR'];
-        if (isset($_SERVER['HTTP_USER_AGENT']))
-            $headers['X-HTTP-User-Agent'] = $_SERVER['HTTP_USER_AGENT'];
-        $headers['Mime-Version'] = '1.0';
-        $headers['Subject'] = qp_encode($headers['Subject']);
-
-        $debug_email = ConfigHelper::getConfig('mail.debug_email');
-        if (!empty($debug_email)) {
-            $recipients = ConfigHelper::getConfig('mail.debug_email');
-            $headers['To'] = '<' . $recipients . '>';
-        }
-
-        if (empty($headers['Date']))
-            $headers['Date'] = date('r');
-
-        if ($files || $headers['X-LMS-Format'] == 'html') {
-            $boundary = '-LMS-' . str_replace(' ', '.', microtime());
-            $headers['Content-Type'] = "multipart/mixed;\n  boundary=\"" . $boundary . '"';
-            $buf = "\nThis is a multi-part message in MIME format.\n\n";
-            $buf .= '--' . $boundary . "\n";
-            $buf .= "Content-Type: text/" . ($headers['X-LMS-Format'] == 'html' ? "html" : "plain") . "; charset=UTF-8\n\n";
-            $buf .= $body . "\n";
-            if ($files)
-                while (list(, $chunk) = each($files)) {
-                    $buf .= '--' . $boundary . "\n";
-                    $buf .= "Content-Transfer-Encoding: base64\n";
-                    $buf .= "Content-Type: " . $chunk['content_type'] . "; name=\"" . $chunk['filename'] . "\"\n";
-                    $buf .= "Content-Description:\n";
-                    $buf .= "Content-Disposition: attachment; filename=\"" . $chunk['filename'] . "\"\n\n";
-                    $buf .= chunk_split(base64_encode($chunk['data']), 60, "\n");
-                }
-            $buf .= '--' . $boundary . '--';
-        } else {
-            $headers['Content-Type'] = 'text/plain; charset=UTF-8';
-            $buf = $body;
-        }
-
-
-        $error = $this->mail_object->send($recipients, $headers, $buf);
-        //if (PEAR::isError($error))
-        if (is_a($error, 'PEAR_Error'))
-            return $error->getMessage();
-        else
-            return MSG_SENT;
+	$persist = is_null($persist) ? ConfigHelper::getConfig('mail.smtp_persist', true) : $persist;
+
+	if(ConfigHelper::getConfig('mail.backend') == 'pear'){
```

Comment:
w upgradedb jest taka linia:
 $this->Execute("INSERT INTO uiconfig (section, var, value, type) VALUES('mail', 'backend', 'pear', ?)", array(CONFIG_TYPE_MAIL_BACKEND));
ona ustawi defaultowo na pear 
ktoś by musiał zapytaniem sql zmienić to w bazie żeby było puste bo w lmsconfigmanager.php w case sprawdzam czy ta wartość jest pear czy phpmailer